### PR TITLE
Label scanner results as 'bad' when versions are blocked (according to the activity log)

### PR DESCRIPTION
**Pull request by willdurand**
Fixes https://github.com/mozilla/addons-server/issues/14086

---

/cc @diox @eviljeff

### CODE REVIEW

File: src/olympia/scanners/views.py [hunk 1]
```diff
@@ -61,7 +61,13 @@ def get_queryset(self):
             .all()
         )
         bad_results = (
-            bad_results.filter(state=TRUE_POSITIVE)
+            bad_results.filter(
+                Q(state=TRUE_POSITIVE) | Q(
+                    version__versionlog__activity_log__action__in=(
+                        amo.LOG.BLOCKLIST_BLOCK_ADDED.id,
```
Comment (eviljeff) [inline src/olympia/scanners/views.py:67]: I think you can add `amo.LOG.BLOCKLIST_BLOCK_EDITED` too.

Comment (willdurand) [reply src/olympia/scanners/views.py:67]: added, thanks

**Comment by willdurand** [comment]
I updated this patch, it should be ready now :)

File: src/olympia/scanners/tests/test_views.py [hunk 2]
```diff
@@ -275,3 +277,30 @@ def test_get_by_label_and_scanner(self):
...
+        )
+        # result labelled as "bad" because the add-on is blocked.
+        blocked_addon_1 = addon_factory()
+        blocked_version_1 = version_factory(addon=blocked_addon_1)
+        ScannerResult.objects.create(scanner=YARA, version=blocked_version_1)
+        block_1 = Block.objects.create(
+            guid=blocked_addon_1.guid, updated_by=self.user
+        )
+        block_activity_log_save(block_1, change=False)
+        # result labelled as "bad" because the add-on is blocked.
```
Comment (diox) [inline src/olympia/scanners/tests/test_views.py:296]: needs to say `and the block has been edited` or something.

File: src/olympia/scanners/tests/test_views.py [hunk 3]
```diff
@@ -275,3 +277,30 @@ def test_get_by_label_and_scanner(self):
...
+        blocked_addon_2 = addon_factory()
+        blocked_version_2 = version_factory(addon=blocked_addon_2)
+        ScannerResult.objects.create(scanner=YARA, version=blocked_version_2)
+        block_2 = Block.objects.create(
+            guid=blocked_addon_2.guid, updated_by=self.user
+        )
+        block_activity_log_save(block_2, change=True)
```
Comment (diox) [inline src/olympia/scanners/tests/test_views.py:303]: Maybe add 2 more cases:
- One with a block added *and* edited.
- One with a true positive and a block added

The idea being to guard against possible regression where we'd return the same add-on multiple times.

**Pull request merged**